compiler/rustc_codegen_ssa/src/mir/mod.rs

```diff
@@ -233,7 +241,8 @@ pub fn codegen_mir<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
         fx.compute_per_local_var_debug_info(&mut start_bx).unzip();
     fx.per_local_var_debug_info = per_local_var_debug_info;
 
-    let traversal_order = traversal::mono_reachable_reverse_postorder(mir, cx.tcx(), instance);
+    let traversal_order: Vec<_> =
+        traversal::reverse_postorder(mir).map(|(block, _data)| block).collect();
```

Comment:
It is important that only mono reachable blocks are codegened. Unreachable blocks may call functions that don't actually get codegened.